pkg/aws/apis/validation/validation.go

```diff
@@ -204,6 +205,27 @@ func validateInstanceMetadata(metadata *awsapi.InstanceMetadataOptions, fldPath
 	return allErrs
 }
 
+func validateCPUOptions(cpuOptions *awsapi.CPUOptions, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cpuOptions == nil {
+		return allErrs
+	}
+
+	if cpuOptions.CoreCount == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("coreCount"), "CoreCount is required"))
+	}
+
+	if cpuOptions.ThreadsPerCore == nil {
+		allErrs = append(allErrs, field.Required(fldPath.Child("threadsPerCore"), "ThreadsPerCore is required"))
+	}
```

Comment:
Just for my own understanding, is it working like that for the API: like you can only specify both or none ?

Reply:
While I haven't actually tried this myself, the documentation states `The CPU options for the instance. Both the core count and threads per core must be specified in the request.`. Defaulting or some smart lookups do not work here as the "defaults" are different for each machine type, so we just require both.